cirq/protocols/apply_channel.py

```diff
@@ -324,3 +433,26 @@ def _apply_krauss_multi_qubit(krauss: Union[Tuple[Any], Sequence[Any]],
                 out=args.target_tensor)
         args.out_buffer += args.target_tensor
     return args.out_buffer
+
+
+def _incorporate_result_into_target(args: 'ApplyChannelArgs',
+                                    sub_args: 'ApplyChannelArgs',
+                                    sub_result: np.ndarray):
+    """Takes the result of calling `_apply_channel_` on `sub_args` and
+    copies it back into `args.target_tensor` or `args.out_buffer` as
+    necessary to return the result of applying the channel to the full args.
+
+    Args:
+        args: The original args.
+        sub_args: A version of `args` with transposed and sliced views of
+            it's tensors.
+        sub_result: The result of calling an object's `_apply_channel_`
+            method on `sub_args`.  A transposed subspace of the desired
+            result.
+
+    Returns: The full result tensor after applying the channel.  Either
+        `args.target_tensor` or `args.out_buffer`.
+    """
+    return _incorporate_result_into_buffer(args.target_tensor, args.out_buffer,
```

Comment:
Interesting that we don't have to use all the fields.

Reply:
The documentation says that `_apply_channel_` isn't allowed to return either `auxiliary_buffer` as the result.